services/callback/src/main/java/app/coronawarn/server/services/callback/registration/RegistrationRunner.java

```diff
@@ -0,0 +1,52 @@
+package app.coronawarn.server.services.callback.registration;
+
+import static app.coronawarn.server.services.callback.HashingUtils.computeSha256Hash;
+
+import app.coronawarn.server.common.federation.client.FederationGatewayClient;
+import app.coronawarn.server.services.callback.config.CallbackServiceConfig;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.boot.ApplicationArguments;
+import org.springframework.boot.ApplicationRunner;
+import org.springframework.core.annotation.Order;
+import org.springframework.stereotype.Component;
+
+/**
+ * This runner registers the callback service at the EFGS.
+ */
+@Component
+@Order(1)
+public class RegistrationRunner implements ApplicationRunner {
```

Comment:
`ApplicationRunner` do work together with long running services? cool!

Reply:
Yes, their original purpose is to execute logic right before the start of the application. Quite a cool concept which we use to run our actual logic once after start.